docs/2.10.0/docs/canton/usermanual/packagemanagement.rst

```diff
@@ -61,235 +61,183 @@ Understanding Package Vetting
+    For most use-cases, the high-level vetting APIs mentioned above should be sufficient.
+    Usage of the lower-level topology APIs can lead to inconsistencies in the participant's topology state
+    and is advised only by expert users.
 
-- The package must not be vetted. This means there shouldn't be an active vetting transaction corresponding to the package.
+Package topology states
+^^^^^^^^^^^^^^^^^^^^^^^
 
-The following tutorial shows how to remove a package using the Canton console. The first step is to upload and identify the
-package ID for the package to be removed.
+With respect to a participant, a package can be in one of the following states:
 
-.. snippet:: package_dar_removal
-    .. success:: val darHash = participant1.dars.upload("dars/CantonExamples.dar")
-    .. success:: val packageId = participant1.packages.find("Iou").head.packageId
+- **Not found** on the participant: The package does not exist in the local participant stores and it can't be referenced in any request to the participant node.
 
-Package removal will initially fail as, by default, uploading the DAR will add a vetting transaction for the package:
+- **Unknown**: The package may exist in the local participant stores, but it has no associated topology transaction issued by the participant node (i.e. it is unknown topology-wise). A package pertaining to a DAR that was uploaded with the vetting flag disabled is unknown.
 
-.. snippet:: package_dar_removal
-    .. failure:: participant1.packages.remove(packageId)
+- **Check-only**: The package appears at least in a `CheckOnlyPackages` topology transaction and it allows a participant to announce that a collection of Daml packages are known but can only be used for validating pre-existing contracts on the ledger and not for executing new Daml transactions. This concept has been introduced in protocol version 7 for supporting :ref:`Smart contract upgrades <smart-contract-upgrades>`.
 
-The vetting transaction must be manually revoked:
+- **Vetted**: A package in this state appears at least in a `VettedPackages` topology transaction and allows the participant to accept new transactions that reference it in Daml action nodes. This state includes is unchanged from the previous protocol versions.
 
-.. snippet:: package_dar_removal
-    .. success(output=3):: val packageIds = participant1.topology.vetted_packages.list().map(_.item.packageIds).filter(_.contains(packageId)).head
-    .. success(output=5):: participant1.topology.vetted_packages.authorize(TopologyChangeOp.Remove, participant1.id, packageIds, force = true)
+For a DAR that is unknown (topology-wise), the vetting operations (``participant.dars.vetting.enable(darHash)`` or ``PackageService.vetDar``) issues a `VettedPackages` topology transaction referencing all the packages in the DAR.
 
-And then the package can be removed:
+To exemplify, let's vet the examples DAR again:
```

Comment:
```suggestion
To illustrate the point, please vet the example DAR again:
```

Reply:
It's exactly the same thing but more verbose, isn't it?